src/server/service/kv.rs

```diff
@@ -1989,14 +1989,14 @@ fn future_prewrite<E: Engine, L: LockManager>(
     options.lock_ttl = req.get_lock_ttl();
     options.skip_constraint_check = req.get_skip_constraint_check();
     options.for_update_ts = req.get_for_update_ts().into();
-    options.is_pessimistic_lock = req.take_is_pessimistic_lock();
     options.txn_size = req.get_txn_size();
     options.min_commit_ts = req.get_min_commit_ts().into();
 
     let (cb, f) = paired_future_callback();
     let res = storage.async_prewrite(
         req.take_context(),
         mutations,
+        req.take_is_pessimistic_lock(),
```

Comment:
Could we combine combine is_pessimistic_lock with mutations here? Just like this:
```
    let mutations = req
        .take_mutations()
        .into_iter().zip(is_pessimistic_lock.iter()
        .map(|mut x, is_lock| match x.get_op() {
            Op::Put => Mutation::Put((Key::from_raw(x.get_key()), x.take_value()), is_lock),
            Op::Del => Mutation::Delete(Key::from_raw(x.get_key()), is_lock),
            Op::Lock => Mutation::Lock(Key::from_raw(x.get_key()), is_lock),
            Op::Insert => Mutation::Insert((Key::from_raw(x.get_key()), x.take_value()), is_lock),
            _ => panic!("mismatch Op in prewrite mutations"),
        })
```

Reply:
I'm doing this in a follow-up PR.